odbc/src/api/functions.rs

```diff
@@ -1702,19 +1702,56 @@ pub extern "C" fn SQLTables(
     unsupported_function(MongoHandleRef::from(statement_handle), "SQLTables")
 }
 
+fn sql_tables(
+    statement_handle: &RwLock<Statement>,
+    catalog: &str,
+    schema: &str,
+    table: &str,
+    _table_t: &str,
+) -> Result<Box<dyn MongoStatement>> {
+    let statement_reader = statement_handle.read().unwrap();
+    match (catalog, schema, table) {
+        ("SQL_ALL_CATALOGS", "", "") => unsafe {
```

Comment:
Why is this in an unsafe block? Is there a way to avoid that?
The original idea was to bound unsafe blocks to the `extern "c" SQLxxxx` functions to mitigate risks around using unsafe blocks. If you can push the unsafe logic into SQLTablesW that'd be great.

Reply:
Okay thanks for the context on this.  I was able to move the unsafe block to the `SQLTablesW()` function.